src/main/java/duke/storage/FileSystem.java

```diff
@@ -0,0 +1,117 @@
+package duke.storage;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exception.InvalidDateTimeException;
+import duke.helper.TaskList;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * FileSystem class that handles the saving and loading of tasks
+ */
+public class FileSystem {
+    private File file;
+
+    /**
+     * Constructor for FileSystem class
+     *
+     * @param relFilePath files's location
+     * @throws IOException If file cannot be created
+     */
+    public FileSystem(String relFilePath) throws IOException {
+        String dirPath = relFilePath.split("/")[0];
+        File dir = new File(dirPath);
+        file = new File(relFilePath);
+
+        if (!dir.exists()) {
+            dir.mkdirs();
+        }
+
+        if (!file.exists()) {
+            file.createNewFile();
+        }
+    }
+
+    /**
+     * Updates the file according to the tasks stored on the list
+     *
+     * @param taskList the TaskList that contains all the tasks that will be stored in the file
+     */
+    public void updateFile(TaskList taskList) {
+        try {
```

Comment:
This can be expensive (rewriting the entire file with each operation) if your task list grows in size. Instead, you may modify specific rows/lines in your file with each read/write operation and keep your interactions with the file *atomic*